File: chapter6/a_vector_type.rb
class Vec
  attr_reader :x, :y

  def initialize(x, y)
    @x = x
    @y = y
  end

  def plus(vector)
    Vec.new(x + vector.x, y + vector.y)
  end

  def minus(vector)
    Vec.new(x - vector.x, y - vector.y)
  end

  def length
    Math.sqrt(x ** 2 + y ** 2)
  end
end

vec1 = Vec.new(3, 4)
vec2 = Vec.new(2, 5)

p vec1.plus(vec2)
p vec2.minus(vec1)

p vec1.length

File: chapter6/a_vector_type.js
class Vec {
  constructor(x, y) {
    this.x = x
    this.y = y
  }

  plus(vector) {
    return new Vec(
      this.x + vector.x,
      this.y + vector.y
      )
  }

  minus(vector) {
    return new Vec(
      this.x - vector.x,
      this.y - vector.y
    )
  }

  get length() {
    return Math.sqrt(this.x ** 2 + this.y ** 2)
  }
}

let vec1 = new Vec(3, 4)
let vec2 = new Vec(1, 2)

console.log(vec1.plus(vec2))
console.log(vec2.minus(vec1))

console.log(vec1.length)
console.log(vec2.length)


